src/main/java/ch/njol/skript/effects/EffReplace.java

```diff
@@ -100,10 +101,10 @@ protected void execute(final Event e) {
 				}
 			this.haystack.change(e, haystack, ChangeMode.SET);
 		} else {
-			for (Inventory inv : (Inventory[])haystack)
-				for (ItemStack item : (ItemStack[]) needles)
-					for (Integer slot : inv.all(item).keySet()){
-						inv.setItem(slot.intValue(), (ItemStack)replacement);
+			for (Inventory inv : (Inventory[]) haystack)
+				for (ItemType item : (ItemType[]) needles)

```

Comment:
Adding an utility method like ItemType#replace might be something worth considering.